### ISSUE TITLE
Non UTF-8 characters make the qrcode invalid

### ISSUE THREAD

**Issue by agijbels** [davidshimjs/qrcodejs#266]
Can be reproduced using 

https://codepen.io/davidshimjs/pen/NdBYrg

In the JS, line nr 12, change to following: 
`	qrcode.makeCode("íadsfasdsöadfas");`

It doesn't draw anything.

**Comment by FlorianJa** [comment]
I observed some strange behaviour regarding this.

`ö` works
`ö1` works
`ö12` does not work
`ö123` works
`ö1234` works
`ö12345` does not work
`ö123456` does not work

it does not matter if the chars after `ö` are `1 2 3` or `a b c` .  Same behaviour for `äÄ` `öÖ` and `üÜ`.

**Comment by jp-ed** [comment]
This issue was fixed by pull request #273. Thanks to 19z.

Looks like this repo is no longer maintained, so you can find the fork [here](https://github.com/19z/qrcodejs-fixUTF8).

**Comment by RMerl** [comment]
The minified version of that fork returns an error here:

```
Uncaught SyntaxError: Unexpected identifier (at qrcode.min.js:1:16790)
```
However downloading the non-minified version and minifying it myself works.  So, looks like the minified version of the fork is corrupted.

(and I can't find a way to contact the fork's author @19z either, so I'm leaving this warning here...)

**Comment by onikaSharm1** [comment]
@davidshimjs 
Please help me clarify some of my doubts regarding the qrcode.js library.

Is there any limitation of this library/cdn?
Can anyone provide me with the pros and cons of the qrcode js library?
I am using this cdn in my code: - "https://cdnjs.cloudflare.com/ajax/libs/qrcodejs/1.0.0/qrcode.min.js
Please help me by providing answers to my question

I was trying to find answers to my doubts regarding the qrcode.js library in its documentation, but I couldn't find answers to my questions anywhere.

**Comment by divinity76** [comment]
`øll` triggers the error.

@agijbels your title makes no sense as your test string is valid UTF-8, maybe you meant "non-ascii characters" or "non-latin characters" ? 

An example of "invalid UTF-8" would be `"\xC0\xAF"` (In UTF-8 \xC0 must be followed by a byte in the range of \x80 to \xBF, hence \xC0\xAF is not valid UTF-8)

**Comment by agijbels** [comment]
Yes, the title does not make sense, but I wasn't able to edit the title? I
meant non-ascii characters.

Met vriendelijke groeten,
Annelies Gijbels


Op ma 15 jul 2024 om 20:18 schreef divinity76 ***@***.***>:
